### PR TITLE
OCT-2130: Home Donations E2E + OCT:2133: Home Transactions E2E

### PR DESCRIPTION
**Pull request by jmikolajczyk**
## Definition of Done

1. [ ] If required, the desciption of your change is added to the [QA changelog](https://www.notion.so/octantapp/Changelog-for-the-QA-d96fa3b411cf488bb1d8d9a598d88281) 
2. [ ] Acceptance criteria are met.
3. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
4. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
5. [ ] Unit tests are added unless there is a reason to omit them.
6. [ ] Automated tests are added when required.
7. [ ] The code is merged.
8. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
9. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: client/cypress/utils/moveTime.ts [hunk 1]
```diff
@@ -112,11 +112,15 @@ export const moveTime = (
     waitForLoadersToDisappear();
     // reload is needed to get updated data in the app
     cy.reload();
+  } else if (moveTo === 'decisionWindowClosed') {
+    return cy.reload();
   }
```
Comment (aziolek) [inline client/cypress/utils/moveTime.ts:117]: Why?

File: client/src/components/Home/HomeGridPersonalAllocation/HomeGridPersonalAllocation.tsx [hunk 2]
```diff
@@ -54,6 +54,7 @@ const HomeGridPersonalAllocation: FC<HomeGridPersonalAllocationProps> = ({ class
         <div className={styles.root}>
           <DoubleValue
             cryptoCurrency="ethereum"
+            dataTest="HomeGridPersonalAllocation--current"
```
Comment (aziolek) [inline client/src/components/Home/HomeGridPersonalAllocation/HomeGridPersonalAllocation.tsx:57]:
```suggestion
            dataTest="HomeGridPersonalAllocation__DoubleValue--current"
```

File: client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts [hunk 3]
```diff
@@ -21,15 +23,24 @@ export default function useEpochTimestampHappenedIn(
...
       request(subgraphAddress, GET_EPOCH_TIMESTAMP_HAPPENED_IN, {
         timestamp: timestampSecondsNumber,
       }),
     queryKey: QUERY_KEYS.epochTimestampHappenedIn(timestampSecondsNumber),
-    select: data => data.epoches[0].epoch,
+    select: data => {
+      if (window.Cypress) {
+        return currentEpoch!;
+      }
+      return data.epoches[0].epoch;
```
Comment (aziolek) [inline client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts:43]: Why?

File: client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts [hunk 4]
```diff
@@ -21,15 +23,24 @@ export default function useEpochTimestampHappenedIn(
...
   const { subgraphAddress } = env;
+  const { data: currentEpoch } = useCurrentEpoch();
+  const { data: isDecisionWindowOpen } = useIsDecisionWindowOpen();
   const timestampSecondsNumber = parseInt(timestampSeconds, 10);
 
   return useQuery<GetEpochTimestampHappenedInQuery, any, number, any>({
-    enabled: !!timestampSecondsNumber,
+    enabled: window.Cypress
+      ? currentEpoch !== undefined && isDecisionWindowOpen !== undefined
+      : !!timestampSecondsNumber,
```
Comment (aziolek) [inline client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts:33]: Why?

Comment (jmikolajczyk) [reply client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts:33]: Timestamp in the App is equal to epoch 0.

File: client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts (hunk 3 above)
Comment (jmikolajczyk) [reply client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts:43]: Timestamp in the App is equal to epoch 0.

**Pull request merged**